homeassistant/components/matrix/__init__.py

```diff
@@ -72,27 +75,51 @@
 )
 
 
-def setup(hass, config):
-    """Set up the Matrix bot component."""
-    config = config[DOMAIN]
+async def async_setup(hass, config):
+    """Set up the Matrix bot server."""
+    matrix_config = config.get(DOMAIN, {CONF_ROOMS: [], CONF_COMMANDS: []})
+    hass.data[DOMAIN] = {
+        CONF_ROOMS: matrix_config[CONF_ROOMS],
+        CONF_COMMANDS: matrix_config[CONF_COMMANDS],
+    }
 
+    if CONF_HOMESERVER not in matrix_config:
+        return True
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_HOMESERVER: matrix_config[CONF_HOMESERVER],
+                CONF_USERNAME: matrix_config[CONF_USERNAME],
+                CONF_PASSWORD: matrix_config[CONF_PASSWORD],
+                CONF_VERIFY_SSL: matrix_config[CONF_VERIFY_SSL],
```

Comment:
Since you marked them all as optional, this can raise `KeyError` if the user didn't specify all 4 fields.